Lambda functions development

mregger opened the pull request.

mregger committed:
added a new folder for the two lambda functions, one script for inbound and one script for outbound data

mregger committed:
Added a few extra lines to test if the IoT service calls the lambda function as predicted

mregger committed:
Added configuration file for iot-rds-integration

mregger committed:
Completed the iot-rds-integration function. Further work: specify sensor subtype automatically, and adjust time to local pei time

mregger committed:
Wrote API query lambda function. Needs to be set up in AWS, and then tested to see how the parameters are passed to the function

mregger committed:
Fixed issues that were causing api to not work.

mregger committed:
Added time, sensor id, type and subtype query parameters to the API

mregger committed:
Added range selection clause, now measurements are returned with geolocation data and fixed some bugs

rjarsenault1101 committed:
updated rds-api to include new boolean flag

rjarsenault1101 commented on lambda-functions/iot-rds-integration-cnf.json:
```diff
@@ -0,0 +1,208 @@
+
```
Delete JSON file

rjarsenault1101 commented on lambda-functions/iot-rds-integration.py:
```diff
@@ -0,0 +1,88 @@
+    return {"body": errmsg , "headers": {}, "statusCode": 400,
+        "isBase64Encoded":"false"}
+
+logger.info("Cold start complete.")
```
delete

rjarsenault1101 commented on lambda-functions/iot-rds-integration.py:
```diff
@@ -0,0 +1,88 @@
+
+def handler(event,context):
+
+    #logger.info('Data received: ', str(event))
```
delete commented out code

rjarsenault1101 commented on lambda-functions/rds-api-cnf.json:
```diff
@@ -0,0 +1,208 @@
+
```
delete JSON

rjarsenault1101 commented on lambda-functions/rds-api.py:
```diff
@@ -0,0 +1,160 @@
+    return {"body": errmsg , "headers": {}, "statusCode": 400,
+        "isBase64Encoded":"false"}
+
+logger.info("Cold start complete.")
```
remove

rjarsenault1101 commented on lambda-functions/rds-api.py:
```diff
@@ -0,0 +1,160 @@
+logger.info("Cold start complete.")
+
+def handler(event,context):
+    #print('parameters: ', event['queryStringParameters'])
```
delete

rjarsenault1101 commented on lambda-functions/rds-api.py:
```diff
@@ -0,0 +1,160 @@
+    '''
+    parameters = event['queryStringParameters']
+    
+    if parameters is not None:
```
Maybe add a comment describing what this cascading if is doing?

rjarsenault1101 commented on lambda-functions/rds-api.py:
```diff
@@ -0,0 +1,160 @@
+    command = command[:-5] + ' ORDER BY reading_time DESC LIMIT ' + str(max_readings) + ';'
+
+    
+    print(command)
```
don't need the print line

mregger committed:
Deleted iot-rds-conf.json file

iot-rds-conf file is useless; the configurations were done directly on aws

mregger committed:
Deleted api configuration file

The file is no longer needed, the configuration was manually entered on the AWS interface

mregger committed:
Deleted log statements and commented code

mregger committed:
Update rds-api.py

mregger commented:
Merging lambda functions development with the development branch. API and IoT-RDS are operational
